Source/DafnyCore/Verifier/ProofDependency.cs

```diff
@@ -66,6 +66,20 @@ public class ProofObligationDependency : ProofDependency {
   }
 }
 
+public class AssumedProofObligationDependency : ProofDependency {
+  public override RangeToken Range { get; }
+
+  public PODesc.ProofObligationDescription ProofObligation { get; }
+
+  public override string Description =>
```

Comment:
This text doesn't occur in new tests. What is it for?

Reply:
It's pretty difficult to cause the fact being assumed here to be necessary to complete a later proof (which is why it was leading to spurious warnings before!). Usually these assumptions come about as part of proving side conditions about partial operations. They usually get proven automatically, pretty easily, and therefore can be re-proved later without relying the assumption that they've already been proved. I've been working on a test that causes this message to show up, but I haven't managed to create one yet.